File: src/BetPlay.Domain/BettingSlip.cs
namespace BetPlay.Domain;

public class BettingSlip : Entity
{
    public BettingSlip()
    {
        Bets = new List<BettingSlipBet>();
    }

    public float Stake { get; set; }
    public float TotalOdds { get; set; }
    public Dto.Bets.BettingSlipStatus Status { get; set; }
    public DateTime Date { get; set; }
    public Guid UserId { get; set; }
    public virtual User User { get; set; }
    public virtual List<BettingSlipBet> Bets { get; set; }
}

public enum BettingSlipStatus
{
    Pending,
    Won,
    Lost
}

File: betplay-web/types/bets/betting-slip.ts
import { BettingSlipBet } from "./betting-slip-bet";

export type BettingSlip = {
  id: string;
  totalStake: number;
  totalOdds: number;
  status: string;
  bets: BettingSlipBet[];
  date: string;
};

export type BettingSlipWithoutBets = {
  id: string;
  totalStake: number;
  totalOdds: number;
  status: number;
  date: string;
};
